File: examples/java/patterns/code-organization/parallelism.java
// A small number of fixed branches.
var enrich = context.parallel("enrich")
    .step("fx", FxRates.class, ctx -> fxRates.latest())
    .step("weather", Weather.class, ctx -> weatherApi.get())
    .step("quote", Quote.class, ctx -> quoteApi.get())
    .run();

// A variable number of items.
BatchResult<Result> processed = context.map(
    "process-items",
    items,
    Result.class,
    (ctx, item) -> process(item));

File: examples/typescript/patterns/code-organization/parallelism.ts
// A small number of fixed branches.
const { fx, weather, quote } = await context.parallel("enrich", [
  async (ctx) => ctx.step("fx", () => fxRates.latest()),
  async (ctx) => ctx.step("weather", () => weatherApi.get()),
  async (ctx) => ctx.step("quote", () => quoteApi.get()),
]);

// A variable number of items.
const results = await context.map(items, async (ctx, item) =>
  ctx.step("process", () => process(item)),
  { maxConcurrency: 10 },
);
